tests/test_unit_tests.py

```diff
@@ -91,6 +102,13 @@ def test_reader_close_cleanup(self):
             self.assertIsNone(reader._own_stream)
             # Verify reader is marked as closed
             self.assertTrue(reader._closed)
+    
+    def test_resource_to_stream_on_closed_reader(self):
```

Comment:
That test is great as is (don't use the with statement) as we exactly want to test that case for the close call here.